lib/ood_core/job/adapters/lsf/helper.rb

```diff
@@ -78,40 +78,40 @@ def parse_cpu_used(cpu_used)
   def batch_submit_args(script, after: [], afterok: [], afternotok: [], afterany: [])
     args = []
 
-    args += ["-P", script.accounting_id] unless script.accounting_id.nil?
-    args += ["-cwd", script.workdir.to_s] unless script.workdir.nil?
-    args += ["-J", script.job_name] unless script.job_name.nil?
-    args[-1] += "[#{script.job_array_request}]" unless script.job_array_request.nil?
-
-    args += ["-q", script.queue_name] unless script.queue_name.nil?
-    args += ["-U", script.reservation_id] unless script.reservation_id.nil?
-    args += ["-sp", script.priority] unless script.priority.nil?
-    args += ["-H"] if script.submit_as_hold
-    args += (script.rerunnable ? ["-r"] : ["-rn"]) unless script.rerunnable.nil?
-    args += ["-b", script.start_time.localtime.strftime("%Y:%m:%d:%H:%M")] unless script.start_time.nil?
-    args += ["-W", (script.wall_time / 60).to_i] unless script.wall_time.nil?
-    args += ["-L", script.shell_path.to_s] unless script.shell_path.nil?
+    args.concat ["-P", script.accounting_id] unless script.accounting_id.nil?
+    args.concat ["-cwd", script.workdir.to_s] unless script.workdir.nil?
+    args.concat ["-J", script.job_name] unless script.job_name.nil?
+    args[-1].concat "[#{script.job_array_request}]" unless script.job_array_request.nil?
```

Comment:
This doesn't work and unfortunately there was no test case to find it. Can you add it in this PR? Specifically that the -J option is modified correctly when given `job_array_request`.

```ruby
irb(main):012:0> args = ['foo']
=> ["foo"]
irb(main):013:0> args[-1] += 'bar'
=> "foobar"
irb(main):014:0> args[-1].concat ['bar']
Traceback (most recent call last):
        3: from /opt/rh/rh-ruby25/root/usr/bin/irb:11:in `<main>'
        2: from (irb):14
        1: from (irb):14:in `concat'
TypeError (no implicit conversion of Array into String)
```